pkg/grafana/alertgroup-handler.go

```diff
@@ -212,18 +212,52 @@ func (h *AlertRuleGroupHandler) updateAlertRule(rule *models.ProvisionedAlertRul
 	return err
 }
 
-func (h *AlertRuleGroupHandler) putAlertRuleGroup(resource grizzly.Resource) error {
-	// TODO: Turn spec into a real models.AlertRuleGroup object
+func unMarshalAlertRuleGroup(resource grizzly.Resource) (*models.AlertRuleGroup, error) {
 	data, err := json.Marshal(resource.Spec())
 	if err != nil {
-		return err
+		return nil, err
 	}
 	var group models.AlertRuleGroup
 	err = json.Unmarshal(data, &group)
 	if err != nil {
-		return err
+		return nil, err
+	}
+	return &group, nil
+}
+
+func updateResourceToHaveUids(existing, resource grizzly.Resource) (*models.AlertRuleGroup, error) {
+	group, err := unMarshalAlertRuleGroup(existing)
+	if err != nil {
+		return nil, err
+	}
+	t := make(map[*string]string)
```

Comment:
This needs to be a `map[string]string`.

By using pointers, the lookup below will always fail as the same value will have a different pointer

Reply:
Done.  (Though it was working in my testing.  But I am no expert on golang pointers)